src/wake/WakeCategoryEventHandler.cpp

```diff
@@ -45,30 +48,151 @@ namespace UKControllerPlugin {
 
         std::string WakeCategoryEventHandler::GetTagItemDescription(int tagItemId) const
         {
-            return "Aircraft Type / UK Wake Category";
+            if (tagItemId == this->tagItemIdAircraftTypeCategory) {
+                return "Aircraft Type / UK Wake Category";
+            } else if (tagItemId == this->tagItemIdStandaloneCategory) {
+                return "UK Wake Category";
+            } else if (tagItemId ==  this->tagItemIdRecat) {
+                return "RECAT-EU Category";
+            } else if (tagItemId == this->tagItemIdUkRecatCombined) {
+                return "UK / RECAT-EU Combined";
+            } else if (tagItemId == this->tagItemIdAircraftTypeRecat) {
+                return "Aircraft Type / RECAT-EU Category";
+            }
+
+            return "";
         }
 
         /*
             Get item from the cache or generate if not exists
         */
         void WakeCategoryEventHandler::SetTagItemData(TagData& tagData)
         {
-            if (this->cache.count(tagData.flightPlan.GetCallsign()) == 0) {
-                std::string tagString = tagData.flightPlan.GetAircraftType()
-                    + "/" + this->mapper.MapFlightplanToCategory(tagData.flightPlan);
-
-                // 15 characters is the max for tag functions, trim the aircraft type accordingly
-                if (tagString.size() > this->maxItemSize) {
-                    const unsigned int charactersToTrim = tagString.size() - this->maxItemSize;
-                    const std::string aircraftType = tagData.flightPlan.GetAircraftType();
-                    tagString = aircraftType.substr(0, aircraftType.size() - charactersToTrim) + "/"
-                        + this->mapper.MapFlightplanToCategory(tagData.flightPlan);
-                }
-
-                this->cache[tagData.flightPlan.GetCallsign()] = tagString;
+            if (tagData.itemCode == this->tagItemIdAircraftTypeCategory) {

```

Comment:
Same here, or a 'map style' object with function pointers/function name maybe? I'm not sure how C++ likes such modern practices :D

Reply:
Made it all modern like!